File: code-python/file/dir.py
import os

# 检查目录是否存在
if os.path.isdir(os.path.dirname(__file__)+"/exampleDir"):
    print("目录存在")
else:
    print("目录不存在")

# 创建目录
try:
    os.mkdir(os.path.dirname(__file__)+"/exampleDir")
    print("目录创建成功")
except FileExistsError:
    print("目录已存在")
except Exception as e:
    print("创建目录失败:", e)

# 打开并遍历目录
try:
    with os.scandir(os.path.dirname(__file__)+"/exampleDir") as entries:
        print("目录内容:")
        for entry in entries:
            print(entry.name)
except FileNotFoundError:
    print("无法打开目录")

# 删除目录
# try:
#     os.rmdir("exampleDir")
#     print("目录删除成功")
# except OSError as e:
#     print("删除目录失败:", e)

File: code-php/file/dir.php
<?php
// 检查目录是否存在
if (is_dir("exampleDir")) {
    echo "目录存在\n";
} else {
    echo "目录不存在\n";
}

// 创建目录
if (mkdir("exampleDir", 0755)) {
    echo "目录创建成功\n";
} else {
    echo "创建目录失败\n";
}

// 打开目录
$dir = opendir("exampleDir");
if ($dir) {
    // 遍历目录
    echo "目录内容:\n";
    while (($file = readdir($dir)) !== false) {
        echo $file . "\n";
    }
    closedir($dir);
} else {
    echo "无法打开目录\n";
}

// 删除目录
// if (rmdir("exampleDir")) {
//     echo "目录删除成功\n";
// } else {
//     echo "删除目录失败\n";
// }
?>
